pkg/admission/prep_downscale.go

```diff
@@ -23,24 +23,33 @@ import (
 
 	"github.com/grafana/rollout-operator/pkg/config"
 	"github.com/grafana/rollout-operator/pkg/util"
+	"github.com/thanos-io/objstore"
 )
 
 const (
 	PrepareDownscaleWebhookPath = "/admission/prepare-downscale"
 )
 
-func PrepareDownscale(ctx context.Context, logger log.Logger, ar v1.AdmissionReview, api *kubernetes.Clientset) *v1.AdmissionResponse {
+func PrepareDownscale(ctx context.Context, logger log.Logger, ar v1.AdmissionReview, api *kubernetes.Clientset, useZoneTracker bool) *v1.AdmissionResponse {
 	client := &http.Client{
 		Timeout: 5 * time.Second,
 	}
-	return prepareDownscale(ctx, logger, ar, api, client)
+
+	var zt *zoneTracker
+	if useZoneTracker {
+		// TODO(jordanrushing): fix bucket creation semantics and wire-in config supporting multiple CSPs
+		bkt := objstore.NewInMemBucket()
+		zt = newZoneTracker(bkt, "testkey")
+	}
+
+	return prepareDownscale(ctx, logger, ar, api, client, zt)
```

Comment:
Like I mentioned in our 1:1, I think the easiest option for now is to turn the `zoneTracker` param into a string. If it's not nil, require that it be one of `gcs`, `s3`, `whatever azure storage is called`, etc. That's probably easier than trying to wire up a config file, configmap that populates that config file, etc. so that there could be a `zoneTracker` storage config in the file that specifies the storage backend.

Reply:
I ended up adding the necessary config with flags passed to the zoneTracker instantiation before passing `zt` through to `prepareDownscale`